### PR TITLE
Add long index support to ZRangeParams (#4445)

### LINKED ISSUE #4445: Deprecation gap: no non-deprecated descending range-with-scores API that supports long start/end indexes
### Summary
  `zrevrangeWithScores(String key, long start, long end)` is deprecated, but the recommended replacement path appears to be `zrangeWithScores(String, ZRangeParams)`, where `ZRangeParams.zrangeParams(...)` only accepts `int`
  indexes.

  This creates a regression for callers that need `long` offsets (very large sorted sets / deep pagination).

  ### Jedis Version
  7.3.0

  ### Problem
  Deprecated API (supports long):
  - `zrevrangeWithScores(String key, long start, long end)`

  Non-deprecated API (descending via REV, but index args are int):
  - `zrangeWithScores(String key, ZRangeParams.zrangeParams(int start, int stop).rev())`

  So migrating off deprecated API can force narrowing `long -> int`, which is unsafe for large indexes.

  ### Suggested fix
  Any of these would resolve the gap:
  1. Add `long` overloads for index-based params construction (e.g. `zrangeParams(long, long)`).
  2. Add non-deprecated descending `withScores` APIs that accept `long` start/end directly.
  3. Keep/de-undeprcate long-based reverse range-with-scores until a full long-capable replacement exists.

  ### Minimal repro
  ```java
  long start = 3_000_000_000L;
  long end = start + 99;

  // Deprecated but long-safe:
  jedisCluster.zrevrangeWithScores("lb:key", start, end);

  // Non-deprecated replacement requires int (cannot represent the same range safely):
  jedisCluster.zrangeWithScores(
      "lb:key",
      ZRangeParams.zrangeParams((int) start, (int) end).rev()
  );

  ### Expected

  A non-deprecated descending range-with-scores path that supports long indexes.

  ### Actual

  Only deprecated API supports long; non-deprecated path constrains to int.
```

Thank you!

### PR DESCRIPTION
**Pull request by ggivo**
Adds `ZRangeParams.zrangeParams(long, long)` overload to support large sorted set indexes, resolving deprecation gap where `zrevrangeWithScores(long, long)` had no non-deprecated replacement for long indexes.

Fixes #4445

---

> [!NOTE]
> **Low Risk**
> Small API extension and constructor delegation change in `ZRangeParams`; risk is limited to potential edge-case differences in argument encoding/equality for range parameters, covered by added tests.
> 
> **Overview**
> Adds `ZRangeParams.zrangeParams(long, long)` (and a `long` constructor), and makes the existing `int` constructor delegate to the `long` path to support large sorted-set index ranges without precision loss.
> 
> Expands test coverage to assert correct argument/RESP serialization for large `long` values, introduces reusable `CommandArgumentsMatchers` + `ProtocolTestUtil` test helpers, updates affected tests/imports, and broadens the formatter include patterns in `pom.xml` accordingly.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 9fb4feb4d06c5a3d6cf1793c6e4f8a58855854e3. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

File: src/test/java/redis/clients/jedis/commands/unified/SortedSetCommandsTestBase.java [hunk 1]
```diff
@@ -323,6 +323,22 @@ public void zrangeParams() {
...
+
+    long min = 0;
+    long max = 1;
+
+    jedis.zadd("foo", 1, "a");
+    jedis.zadd("foo", 2, "b");
+
+    List<String> expected = new ArrayList<String>();
+    expected.add("b");
+    expected.add("a");
```
[the remark this answers is not in the document]
Comment (ggivo) [reply src/test/java/redis/clients/jedis/commands/unified/SortedSetCommandsTestBase.java:339]: addressed with unit test
